### ISSUE TITLE
Links that match collection names are getting replaced in GraphQL query

### ISSUE THREAD

**Issue by xveganxxxedgex** [gatsbyjs/gatsby#22414]
## Description

We have a project using netlify-cms and have an issue where frontmatter links are getting manipulated when returned by the GraphQL queries when the values match a collection name. It doesn't seem to affect markdown content, just frontmatter values.

We have an `about` collection that supports each site in our monorepo to have their own About page. If we try to link to `/about` from another collection (like if we have a `navbar` collection with a list of links that accept a string for each URL), the URL will get altered and does not return correctly from the graphQL query.

**Collection Data File**

See the `link` value is `/about`. This is the correct value.

![data about](https://user-images.githubusercontent.com/4555168/77109031-7b460a80-69e0-11ea-9e9d-e2e9de9d651f.png)

**GraphQL Result**

Here, the query altered the `link` field, which is not desired:

![about](https://user-images.githubusercontent.com/4555168/77109046-7f722800-69e0-11ea-8a57-ff604f1fd30a.png)

It only seems to do it if the field does not have a subpath, when I tried a `/about/test` value, it did not alter the value:

**Collection Data File**

![Screen Shot 2020-03-19 at 9 57 33 AM](https://user-images.githubusercontent.com/4555168/77093338-11ba0200-69c8-11ea-9c32-c7cb4d8cb6d3.png)

**GraphQL Result**

![:about:test](https://user-images.githubusercontent.com/4555168/77093348-14b4f280-69c8-11ea-8f41-282c7886a901.png)

These examples are test cases used while editing a `career` collection we have. If I change the `link` url to be the same name as the collection it's in (so in this case `/career`), it removes the link entirely: 

**Collection Data File**

![data_career](https://user-images.githubusercontent.com/4555168/77109176-b5171100-69e0-11ea-8ac4-c169d68d1b47.png)

**GraphQL Result**

![career](https://user-images.githubusercontent.com/4555168/77109139-a597c800-69e0-11ea-945f-273c57c3b12f.png)

This is the plugin config for our base gatsby repo (or theme?) `gr8-gatsby`:

```
module.exports.remarkTransformOptions = {
  // CommonMark mode (default: true)
  commonmark: true,
  // Footnotes mode (default: true)
  footnotes: true,
  // Pedantic mode (default: true)
  pedantic: true,
  // GitHub Flavored Markdown mode (default: true)
  gfm: true,
};

module.exports = () => ({
  plugins: [
    {
      resolve: "gatsby-source-filesystem",
      options: {
        name: "images",
        path: `${__dirname}/images`,
      },
    },
    `gatsby-transformer-sharp`, // run these plugins or gatsby-image breaks your build
    `gatsby-plugin-sharp`, // run these plugins or gatsby-image breaks your build
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `markdown`,
        path: `${__dirname}/data`,
      },
    },
    {
      resolve: `gatsby-transformer-remark`,
      options: Object.assign({}, module.exports.remarkTransformOptions, {
        // Plugins configs
        plugins: [
          {
            resolve: "gatsby-remark-component",
            options: { components: [] },
          },
        ],
      }),
    },
    {
      resolve: `gatsby-plugin-styled-components`,
      options: { displayName: true },
    },
  ],
});
```

Then in each individual site we have:

```
module.exports = {
  siteMetadata,
  plugins: [
    {
      resolve: "gr8-gatsby", // our internal shared component repo/theme that has the above config
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-plugin-netlify-cms`,
      options: {
        enableIdentityWidget: false,
        modulePath: "./cms.js",
        manualInit: true,
      },
    },
    `gatsby-plugin-styled-components`,
  ],
};
```

I've tried looking through the gatsby, netlify, and various plugins code to see if I can find where/why this manipulation is happening, but I haven't been able to track it down. Does this seem like expected behaviour with how collections and URLs are handled? 

### Steps to reproduce

I'm trying to get a test repo going but am running into issues getting the build to pass. If I can get it working, I'll update this with a link.

These would be the replication steps on our project where it's broken:

* Make a collection for `about`

```
export const about = {
  label: "About",
  name: "about",
  folder: "data/about",
  create: true,
  summary: "{{title}}",
  slug: "{{site}}/about",
  fields: [
    {
      label: "Title",
      name: "title",
      widget: "string",
    },
    { label: "Body", name: "body", widget: "markdown" },
  ],
};
```

* Make a collection for `navbar`

```
export const navbar = {
  label: "Navbar",
  name: "navbar",
  folder: "data/navbar",
  create: true,
  summary: "{{title}}",
  fields: [
    {
      label: "Title",
      name: "title",
      widget: "string",
    },
    {
      label: "Links",
      name: "links",
      widget: "list",
      fields: [
        { label: "Title", name: "title", widget: "string" },
        {
          label: "URL",
          name: "url",
          widget: "string",
        },
      ],
    },
  ],
};
```

* Add an entry for `about`
* Add an entry for `navbar` where one of the links has `/about` as the `url` value.
* Your `data` file for the `navbar` should show the `url` value as `/about` as expected
* Open `/__graphql` in the browser and run a query to get the navbar entries
* The `url` field returns `../about` instead of `/about`

### Expected result

Frontmatter URLs should not be converted to relative paths for values that match collection names.

### Actual result

Frontmatter URLs get manipulated by GraphQL queries to be relative paths.

### Environment

```
System:
    OS: macOS 10.15.3
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.1 - /var/folders/6l/z9xsxrqd1y50vpdsfjd189w0tgx7q7/T/yarn--1584635975554-0.7094580855450245/node
    Yarn: 1.17.3 - /var/folders/6l/z9xsxrqd1y50vpdsfjd189w0tgx7q7/T/yarn--1584635975554-0.7094580855450245/yarn
    npm: 6.10.2 - ~/.npm-global/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 80.0.3987.132
    Firefox: 71.0
    Safari: 13.0.5
```

Thank you so much for any help or insight!

**Comment by xveganxxxedgex** [comment]
Here's my replication repo:

https://github.com/xveganxxxedgex/gatsby-starter-netlify-cms

Here, I have a title of `/blog`, where `blog` is the name of the collection. You can see the data file with the `title: '/blog'` here:

https://github.com/xveganxxxedgex/gatsby-starter-netlify-cms/blob/master/src/pages/blog/2020-03-23-blog.md

But query manipulates the title, causing no title to appear in the actual HTML output for that page:

https://objective-mahavira-48dbee.netlify.com/blog/2020-03-23-blog/

![Screen Shot 2020-03-23 at 12 52 13 PM](https://user-images.githubusercontent.com/4555168/77357344-37674400-6d05-11ea-8557-090bb049d5df.png)

**Comment by xveganxxxedgex** [comment]
Not stale

**Comment by xveganxxxedgex** [comment]
Not stale

**Comment by xveganxxxedgex** [comment]
Not stale

**Comment by ironng** [comment]
hey @xveganxxxedgex I ran into a similar issue today, and I think the cause seems to be the method [`fmImagesToRelative`](https://github.com/danielmahon/gatsby-remark-relative-images#to-convert-frontmatter-images) from the [gatsby-remark-relative-images](https://github.com/danielmahon/gatsby-remark-relative-images), which is used to convert  absolute paths for frontmatter images to relative paths. You might want to try commenting out [this line](https://github.com/xveganxxxedgex/gatsby-starter-netlify-cms/blob/master/gatsby-node.js#L77) in your demo repo in the gatsby-node.js file. I just excluded the node I needed from this transformation.
